From f5a23f4b877149d666698f3f18d01b4043a29948 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:43 -0400
Subject: [infiniband] cxgb4: Fix four byte info leak in c4iw_create_cq()

Message-id: <1410475447-94392-25-git-send-email-svemuri@redhat.com>
Patchwork-id: 93889
O-Subject: [RHEL7.1 PATCH BZ 1124947 024/108] RDMA/cxgb4: Fix four byte info leak in c4iw_create_cq()
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

There is a four byte hole at the end of the "uresp" struct after the
->qid_mask member.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit e24a72a3302a638d4c6e77f0b40c45cc61c3f089)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cq.c b/drivers/infiniband/hw/cxgb4/cq.c
index 88de3aa..e436ead 100644
--- a/drivers/infiniband/hw/cxgb4/cq.c
+++ b/drivers/infiniband/hw/cxgb4/cq.c
@@ -930,6 +930,7 @@ struct ib_cq *c4iw_create_cq(struct ib_device *ibdev, int entries,
   if (!mm2)
    goto err4;
 
+  memset(&uresp, 0, sizeof(uresp));
   uresp.qid_mask = rhp->rdev.cqmask;
   uresp.cqid = chp->cq.cqid;
   uresp.size = chp->cq.size;
-- 
1.7.1